common/utils.ts

```diff
@@ -17,21 +18,88 @@ import {
   DATE_FORMAT_PATTERN,
   COMPONENT_CATEGORY,
   NODE_CATEGORY,
+  WorkspaceFormValues,
 } from './';
 
 // TODO: implement this and remove hardcoded return values
 /**
- * Converts a ReactFlow workspace flow to a backend-compatible set of ingest and/or search sub-workflows,
- * along with a provision sub-workflow if resources are to be created.
+ * Given a ReactFlow workspace flow and the set of current form values within such flow,
+ * generate a backend-compatible set of sub-workflows.
+ *
  */
 export function toTemplateFlows(
-  workspaceFlow: WorkspaceFlowState
+  workspaceFlow: WorkspaceFlowState,
+  formValues: WorkspaceFormValues
 ): TemplateFlows {
+  const textEmbeddingTransformerNodeId = Object.keys(formValues).find((key) =>
+    key.includes('text_embedding')
+  ) as string;
+  const knnIndexerNodeId = Object.keys(formValues).find((key) =>
+    key.includes('knn')
+  ) as string;
+  const textEmbeddingFields = formValues[textEmbeddingTransformerNodeId];
+  const knnIndexerFields = formValues[knnIndexerNodeId];
+
   return {
     provision: {
-      user_params: {} as Map<string, any>,
-      nodes: [],
-      edges: [],
+      nodes: [
+        {
+          id: 'create_ingest_pipeline',
+          type: 'create_ingest_pipeline',
+          user_inputs: {
+            pipeline_id: 'test-pipeline',
+            model_id: textEmbeddingFields['modelId'],
+            input_field: textEmbeddingFields['inputField'],
+            output_field: textEmbeddingFields['vectorField'],
+            configurations: {
+              description: 'A text embedding ingest pipeline',
+              processors: [
+                {
+                  text_embedding: {
+                    model_id: textEmbeddingFields['modelId'],
+                    field_map: {
+                      [textEmbeddingFields['inputField']]:
+                        textEmbeddingFields['vectorField'],
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        },
+        {
+          id: 'create_index',
+          type: 'create_index',
+          previous_node_inputs: {
+            create_ingest_pipeline: 'pipeline_id',
+          },
+          user_inputs: {
+            index_name: knnIndexerFields['indexName'],
+            configurations: {
+              settings: {
+                default_pipeline: '${{create_ingest_pipeline.pipeline_id}}',
+              },
+              mappings: {
+                properties: {
+                  [textEmbeddingFields['vectorField']]: {
+                    type: 'knn_vector',
+                    dimension: 768,
+                    method: {
+                      engine: 'lucene',
+                      space_type: 'l2',
+                      name: 'hnsw',
+                      parameters: {},
+                    },
```

Comment:
Why do we have static values defined here? Aren't we taking the inputs from the user or future plans of doing so?

Reply:
same as above. As a reminder in the description, this is all hardcoded right now for the purpose of getting an initial use case possible